src/mem/cache_padded.rs

```diff
@@ -1,48 +1,32 @@
-use std::marker;
+//! Cache-padded values.
+
 use std::cell::UnsafeCell;
-use std::fmt;
-use std::mem;
-use std::ptr;
 use std::ops::{Deref, DerefMut};
+use std::{ptr, mem, fmt, marker};
 
-// For now, treat this as an arch-independent constant.
-const CACHE_LINE: usize = 32;
-
-#[cfg_attr(feature = "nightly",
-           repr(simd))]
-#[derive(Debug)]
-struct Padding(u64, u64, u64, u64);
-
-/// Pad `T` to the length of a cacheline.
-///
-/// Sometimes concurrent programming requires a piece of data to be padded out
-/// to the size of a cacheline to avoid "false sharing": cachelines being
-/// invalidated due to unrelated concurrent activity. Use the `CachePadded` type
-/// when you want to *avoid* cache locality.
-///
-/// At the moment, cache lines are assumed to be 32 * sizeof(usize) on all
-/// architectures.
+/// The size of a cache line.
 ///
-/// **Warning**: the wrapped data is never dropped; move out using `ptr::read`
-/// if you need to run dtors.
-pub struct CachePadded<T> {
-    data: UnsafeCell<[usize; CACHE_LINE]>,
-    _marker: ([Padding; 0], marker::PhantomData<T>),
-}
+/// Do not let your code's safety or correctness depend on this being the real size. The actual
+/// value may not be equal to this.
+// TODO: For now, treat this as an arch-independent constant.
+const CACHE_LINE: usize = 64;
```

Comment:
In x86-64, the size of `CachePadded<T>` changed from 256 (`sizeof(usize) * CACHE_LINE` = 8 * 32 = 256) to 64 (`sizeof(u8) * CACHE_LINE = 1 * 64 = 64).  But 64 bytes are not sufficient: e.g. `Participant`, which is fed to `CachePadded`, is 104 bytes big.  That causes all the test failures.

The tests still fail even after fixing this by setting `const CACHE_LINE: usize = 256;`, though..